File: python/problem_002.py
def fibs_to(lim):
    fibs = [1, 2]
    while sum(fibs[-2:]) <= lim:
        fibs.append(fibs[-1] + fibs[-2])
    return fibs


def main():
    print sum(filter(lambda x: x % 2 == 0, fibs_to(4000000)))


if __name__ == '__main__':
    main()

File: rust/problem_002.rs
fn fibs_to(n: u32) -> Vec<u32> {
    let mut fibs = Vec::with_capacity(2000);
    fibs.push(1);
    fibs.push(2);
    let mut last = 1;

    while fibs[last] < n {
        let (j, k) = (fibs[last], fibs[last-1]);
        fibs.push(j + k);
        last = last + 1;
    }

    fibs
}

fn main() {
    let fibs = fibs_to(4_000_000);
    let sum = fibs.iter().filter(|&n| *n % 2 == 0)
        .fold(0, |a,&b| a + b);

    println!("{}", sum);
}
